### PR TITLE
PLATFORM-2466 adding sjc-shields-healthy headers logging

### PR DESCRIPTION
**Pull request by pgrabowski-wikia**
https://wikia-inc.atlassian.net/browse/PLATFORM-2466

@Grunny @sqreek

### CODE REVIEW

File: lib/Wikia/src/Tracer/WikiaTracer.php [hunk 1]
```diff
@@ -296,6 +300,7 @@ public function getPublicHeaders() {
 			self::TRACE_ID_HEADER_NAME => $this->traceId,
 			self::LEGACY_TRACE_ID_HEADER_NAME => $this->traceId, // duplicated until we move to X-Trace-Id everywhere
```
Comment (WikiaTech) [inline lib/Wikia/src/Tracer/WikiaTracer.php]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

File: lib/Wikia/src/Tracer/WikiaTracer.php [hunk 2]
```diff
@@ -296,6 +300,7 @@ public function getPublicHeaders() {
...
 			self::PARENT_SPAN_ID_HEADER_NAME => $this->spanId, // pass the current span ID to the subrequest, it will be logged as parent_span_id there
```
Comment (WikiaTech) [inline lib/Wikia/src/Tracer/WikiaTracer.php]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* MAJOR 3 major
* MINOR 3 minor
* INFO 7 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WikiaTracer.php#L7](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L7): Class "WikiaTracer" has 28 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WikiaTracer.php#L356](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L356): Remove the unused function parameter "$cmd". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [WikiaTracer.php#L377](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L377): Remove the unused function parameter "$method". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MINOR [WikiaTracer.php#L9](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L9): Define a constant instead of duplicating this literal "mediawiki" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [WikiaTracer.php#L146](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L146): Define a constant instead of duplicating this literal "REQUEST_URI" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [WikiaTracer.php#L153](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L153): Define a constant instead of duplicating this literal "SERVER_NAME" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. INFO [WikiaTracer.php#L113](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L113): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)
1. INFO [WikiaTracer.php#L133](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L133): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)
1. INFO [WikiaTracer.php#L134](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L134): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)
1. INFO [WikiaTracer.php#L135](https://github.com/Wikia/app/blob/92716bb96c6bf8c2f7fbb69e0f64520fc222cd75/lib/Wikia/src/Tracer/WikiaTracer.php#L135): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WikiaTracer.php#L7](https://github.com/Wikia/app/blob/9230c8ae473e3d870874b1e6678560626fcaa3f4/lib/Wikia/src/Tracer/WikiaTracer.php#L7): Class "WikiaTracer" has 28 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WikiaTracer.php#L363](https://github.com/Wikia/app/blob/9230c8ae473e3d870874b1e6678560626fcaa3f4/lib/Wikia/src/Tracer/WikiaTracer.php#L363): Remove the unused function parameter "$cmd". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [WikiaTracer.php#L384](https://github.com/Wikia/app/blob/9230c8ae473e3d870874b1e6678560626fcaa3f4/lib/Wikia/src/Tracer/WikiaTracer.php#L384): Remove the unused function parameter "$method". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MINOR [WikiaTracer.php#L150](https://github.com/Wikia/app/blob/9230c8ae473e3d870874b1e6678560626fcaa3f4/lib/Wikia/src/Tracer/WikiaTracer.php#L150): Define a constant instead of duplicating this literal "REQUEST_URI" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [WikiaTracer.php#L157](https://github.com/Wikia/app/blob/9230c8ae473e3d870874b1e6678560626fcaa3f4/lib/Wikia/src/Tracer/WikiaTracer.php#L157): Define a constant instead of duplicating this literal "SERVER_NAME" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WikiaTracer.php#L7](https://github.com/Wikia/app/blob/8eb270f9b9663f3599c8ec757664c1e17b9d304f/lib/Wikia/src/Tracer/WikiaTracer.php#L7): Class "WikiaTracer" has 28 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WikiaTracer.php#L365](https://github.com/Wikia/app/blob/8eb270f9b9663f3599c8ec757664c1e17b9d304f/lib/Wikia/src/Tracer/WikiaTracer.php#L365): Remove the unused function parameter "$cmd". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [WikiaTracer.php#L386](https://github.com/Wikia/app/blob/8eb270f9b9663f3599c8ec757664c1e17b9d304f/lib/Wikia/src/Tracer/WikiaTracer.php#L386): Remove the unused function parameter "$method". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

File: lib/Wikia/src/Tracer/WikiaTracer.php [hunk 3]
```diff
@@ -36,6 +38,8 @@ class WikiaTracer {
 	private $appVersion = '';
 
 	private $requestPath = [];
+	const SERVER_REQUEST_URI = 'REQUEST_URI';
+	const SERVER_SERVER_NAME = 'SERVER_NAME';
```
Comment (Grunny) [inline lib/Wikia/src/Tracer/WikiaTracer.php]: Minor: I'd move these up with the other constants to keep them grouped.

**Review by Grunny — APPROVED** [review]
Other than the minor comment about the constants, LGTM.

File: lib/Wikia/src/Tracer/WikiaTracer.php (hunk 3 above)
Comment (pgrabowski-wikia) [reply lib/Wikia/src/Tracer/WikiaTracer.php]: I blame PhpStorm :P done

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [WikiaTracer.php#L7](https://github.com/Wikia/app/blob/04df78dae503c92bbf74a16e681b6567d13f0ac5/lib/Wikia/src/Tracer/WikiaTracer.php#L7): Class "WikiaTracer" has 28 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. MAJOR [WikiaTracer.php#L366](https://github.com/Wikia/app/blob/04df78dae503c92bbf74a16e681b6567d13f0ac5/lib/Wikia/src/Tracer/WikiaTracer.php#L366): Remove the unused function parameter "$cmd". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
3. MAJOR [WikiaTracer.php#L387](https://github.com/Wikia/app/blob/04df78dae503c92bbf74a16e681b6567d13f0ac5/lib/Wikia/src/Tracer/WikiaTracer.php#L387): Remove the unused function parameter "$method". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

**Pull request merged**